frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -116,127 +116,37 @@ end
-            args = -1,
-        }
-        if Device:hasKeyboard() then
-            self.key_events.MoveUp = {
-                { "Shift", "RPgBack" },
-                event = "Panning",
-                args = {0, -1},
-            }
-            self.key_events.MoveDown = {
-                { "Shift", "RPgFwd" },
-                event = "Panning",
-                args = {0,  1},
-            }
-        end
-    elseif Device:hasKeys() then
-        self.key_events.GotoNextView = {
-            { { "RPgFwd", "LPgFwd", "Right" } },
-            event = "GotoViewRel",
-            args = 1,
-        }
-        self.key_events.GotoPrevView = {
-            { { "RPgBack", "LPgBack", "Left" } },
-            event = "GotoViewRel",
-            args = -1,
-        }
-    end
     if Device:hasDPad() and Device:useDPadAsActionKeys() then
-        self.key_events.GotoNextChapter = {
-            { "Right" },
-            event = "GotoNextChapter",
-            args = 1,
-        }
-        self.key_events.GotoPrevChapter = {
-            { "Left" },
-            event = "GotoPrevChapter",
-            args = -1,
-        }
+        if G_reader_settings:isTrue("left_right_keys_turn_pages") then
+            self.key_events.GotoNextView = { { { "LPgFwd", "Right" } }, event = "GotoViewRel", args = 1, }
+            self.key_events.GotoPrevView = { { { "LPgBack", "Left" } }, event = "GotoViewRel", args = -1, }
+        elseif G_reader_settings:nilOrFalse("left_right_keys_turn_pages") then
+            self.key_events.GotoNextChapter = { { "Right" }, event = "GotoNextChapter", args = 1, }
+            self.key_events.GotoPrevChapter = { { "Left" }, event = "GotoPrevChapter", args = -1, }
+            self.key_events.GotoNextView = { { "LPgFwd" }, event = "GotoViewRel", args = 1, }
+            self.key_events.GotoPrevView = { { "LPgBack" }, event = "GotoViewRel", args = -1, }
+        end
+        self.key_events.MoveUp = { { "RPgBack" }, event = "Panning", args = {0, -1}, }
+        self.key_events.MoveDown = { { { "RPgFwd", " " } }, event = "Panning", args = {0,  1}, }
     elseif Device:hasDPad() then
-        self.key_events.MoveUp = {
-            { "Up" },
-            event = "Panning",
-            args = {0, -1},
-        }
-        self.key_events.MoveDown = {
-            { "Down" },
-            event = "Panning",
-            args = {0,  1},
-        }
+        self.key_events.MoveUp = { { "Up" }, event = "Panning", args = {0, -1}, }
+        self.key_events.MoveDown = { { "Down" }, event = "Panning", args = {0,  1}, }
     end
-    if Device:hasScreenKB() then
-        self.key_events.MoveUp = {
-            { "ScreenKB", "RPgBack" },
-            event = "Panning",
-            args = {0, -1},
-        }
-        self.key_events.MoveDown = {
-            { "ScreenKB", "RPgFwd" },
-            event = "Panning",
-            args = {0,  1},
-        }
+    if Device:hasKeys() and not (Device:hasScreenKB() or Device:hasSymKey()) then
```

Comment:
```suggestion
    if Device:hasKeys() and not Device:useDPadAsActionKeys() then
```

Reply:
@Frenzie right?

Reply:
In that case it seems it should at least be something like `if (Device:hasDPad() and not Device:useDPadAsActionKeys()) or (Device:hasKeys() and not Device:useDPadAsActionKeys())` if there isn't an obvious better way to do it. 

Duplicating a few lines is only to be avoided as a rule of thumb btw. Sometimes things get much muddier when trying to avoid repeating things.